lesson-8/Ioan-Juglea/Classes Homework.py

```diff
@@ -0,0 +1,52 @@
+class Produs(object):
+    def __init__(self, denumire, pret, cantitate, owner, an_productie):
+        self.denumire = denumire
+        self.pret = pret
+        self.cantitate = cantitate
+        self.owner = owner
+        self.an_productie = an_productie
+
+    def aplica_reducere(self, procent):
+        self.pret = self.pret - self.pret / 100 * procent
+    
+    def restocare(self, numar):
```

Comment:
O varianta aici ar fi sa faci check-ul la if invers ca sa putem iesi din functie mai repede daca conditia nu este favorabila. In cazul asta nu este o imbunatatire majora dar in general ii un principiu bun. Ceva de genu:
```
if numar < 0:
   print(f'{numar} nu este un numar valid')
   return None
self.cantitate +=self.cantitate

```